includes/Core/Util/Entity_Factory.php

```diff
@@ -60,8 +60,15 @@ public static function from_context() {
 
 		// Otherwise, run frontend-specific `WP_Query` logic.
 		if ( $wp_the_query instanceof WP_Query ) {
-			return self::from_wp_query( $wp_the_query );
+			$entity = self::from_wp_query( $wp_the_query );
+
+			if ( is_null( $entity ) || ! defined( 'AMP__VERSION' ) ) {
+				return $entity;
+			}
```

Comment:
Looks like this condition repeats in both methods. Let's move it to the `maybe_convert_to_amp_entity` method and put it at the very beginning of the method:

```php
private static function maybe_convert_to_amp_entity( $url, $entity ) {
    if ( is_null( $entity ) || ! defined( 'AMP__VERSION' ) ) {
        return $entity;
    }

    ...
}
```